owasp-top10-2017-apps/a5/ecommerce-api/app/handlers/handlers.go

```diff
@@ -13,14 +14,40 @@ func HealthCheck(c echo.Context) error {
 	return c.String(http.StatusOK, "WORKING\n")
 }
 
+func ErrorMessage(c echo.Context) error {
+	return c.JSON(http.StatusBadRequest, map[string]string{"result": "error", "details": "Error finding this UserID."})
+}
+
 // GetTicket returns the userID ticket.
 func GetTicket(c echo.Context) error {
+
+	cookie, err := ReadCookie(c)
+
+	if err != nil {
+		return ErrorMessage(c)
+	}
+
+	type UserClaims struct {
```

Comment:
Not in scope of this review, but you could have declared this struct before `HealthCheck` function and then create a new `UserClaims` here 😃: 

```golang
userClaims := UserClaims{}
```